wpimath/src/main/native/include/frc/MathUtil.h

```diff
@@ -12,14 +14,62 @@
 namespace frc {
 
 /**
- * Returns 0.0 if the given value is within the specified range around zero.
- * The remaining range between the deadband and 1.0 is scaled from 0.0 to 1.0.
+ * Returns 0.0 if the given value is within the specified range around zero. The
+ * remaining range between the deadband and the maximum magnitude is scaled from
+ * 0.0 to the maximum magnitude (or left unchanged if the maximum magnitude is
+ * infinite).
  *
- * @param value    Value to clip.
+ * @param value Value to clip.
  * @param deadband Range around zero.
+ * @param maxMagnitude The maximum magnitude of the input (defaults to 1).
+ * @return The value after the deadband is applied.
  */
-WPILIB_DLLEXPORT
-double ApplyDeadband(double value, double deadband);
+template <typename T>
+T ApplyDeadband(T value, T deadband, T maxMagnitude = T{1}) {
+  if (std::fabs(static_cast<double>(value)) > static_cast<double>(deadband)) {
```

Comment:
```suggestion
  if (std::abs(value) > deadband) {
```

Reply:
I don't think this works if T is a unit type

Reply:
Use `if constexpr` to differentiate between types if they each need to do something different.